internal/dataplane/kongstate/kongstate.go

```diff
@@ -756,93 +791,107 @@ func (ks *KongState) fetchEntitySchema(schemaGetter SchemaGetter, entityType str
 	return ExtractEntityFieldDefinitions(schema), nil
 }
 
-// fillCustomEntityForeignFields fills the "foreign" fields of a custom Kong entity
-// if it refers to a service/route/consumer.
-// Because Kong gateway requires the "foreign" fields to use IDs of the referred entity,
-// it fills ID of the referred entity and fill the ID of the entity to the field.
-// So the function has the side effect that the referred entity will have a generated fixed ID.
-// It does not support fields referring to other types of entities.
-func (ks *KongState) fillCustomEntityForeignFields(
-	logger logr.Logger,
-	k8sEntity *kongv1alpha1.KongCustomEntity,
-	schema EntitySchema,
-	parsedEntity map[string]any,
-	pluginRelEntities PluginRelatedEntitiesRefs,
-	workspace string,
-) {
-	logger = logger.WithValues("entity_namespace", k8sEntity.Namespace, "entity_name", k8sEntity.Name)
+func (ks *KongState) findCustomEntityRelatedPlugin(k8sEntity *kongv1alpha1.KongCustomEntity) (string, bool) {
 	// Find referred entity via the plugin in its spec.parentRef.
 	// Then we can fetch the referred service/route/consumer from the reference relations of the plugin.
 	parentRef := k8sEntity.Spec.ParentRef
 	var namespace string
 	// Abort if the parentRef is empty or does not refer to a plugin.
 	if parentRef == nil ||
 		(parentRef.Group == nil || *parentRef.Group != kongv1alpha1.GroupVersion.Group) {
-		return
+		return "", false
 	}
 	if parentRef.Kind == nil || (*parentRef.Kind != "KongPlugin" && *parentRef.Kind != "KongClusterPlugin") {
-		return
+		return "", false
 	}
 	// Extract the plugin key to get the plugin relations.
 	if parentRef.Namespace == nil || *parentRef.Namespace == "" {
 		namespace = k8sEntity.Namespace
 	} else {
 		namespace = *parentRef.Namespace
 	}
-	pluginKey := namespace + ":" + parentRef.Name
+	return namespace + ":" + parentRef.Name, true
+}
+
+func (ks *KongState) findCustomEntityForeignFields(
```

Comment:
I think this function could be covered with unit tests for every possible combination.

Reply:
Done.